py2neo/cypher/core.py

```diff
@@ -168,6 +173,41 @@ def execute_one(self, statement, parameters=None, **kwparameters):
             finally:
                 response.close()
 
+    def execute_for_visualisation(
+            self, statement, parameters=None, **kwparameters):
+        """ Execute a single cypher query and return the result in graphJSON
+        format ready for visualisation.
+
+        The format collates all the nodes and relationships from all columns of
+        the result, and also flattens collections of nodes and relationships,
+        including paths.
+
+        """
+        if self.transaction_uri:
```

Comment:
i'd expect callers of this not to be running queries that will change the graph, so possibly i can ignore this path? the docstring could make it clear that this is expected to be treated as a read api only and so it does not run in a transaction. @nigelsmall, what do you think?